WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -225,8 +225,9 @@ def post_results
         body += "#{record_strs.join(", ")}.  \n" # Trailing spaces for markdown give us a <br>
       end
     end
-    post = Post.new(title: title, body: body, author: current_user, world_readable: true)
-    post_post(post)
+    comp.update(results_posted_at: Time.now)
+    comp.save!
```

Comment:
**Ugly page:** Agreed. I was leaning in the other direction, that maybe we should have a nice page for that failure. But I think that's overkill.

**Posting results:** Thanks! That makes a ton of sense! Only showing results publicly when complete sounds both important (incomplete results making the wrong person look like world champion for some time?) and easy. `publish_results()` maybe?

You can also imagine a change log on the page if results are edited after publishing.